samples/drivers/adc/src/main.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/printk.h>
+#include <drivers/adc.h>
+
+/*
+ * The used ADC is specified in the devicetree using adc0 alias. This will
+ * automatically enable the sample for that board.
+ */
+#define ADC_NODE DT_ALIAS(adc0)
+
+#if DT_NODE_HAS_STATUS(ADC_NODE, okay)
+#define ADC_DEVICE_NAME		DT_LABEL(ADC_NODE)
+#else
+#error "Unsupported board: ADC devicetree alias not defined"
+#endif
+
+/*
+ * Read sequence of two channels for demonstration (if supported by MCU and
+ * driver)
+ */
+#if defined(CONFIG_ADC_STM32) || \
+	defined(CONFIG_ADC_SAM0)
+#define ADC_NUM_CHANNELS	1
+#else
+#define ADC_NUM_CHANNELS	2
+#endif
+
+/*
+ * Common settings supported by most ADCs. The adc0 alias should be added to
+ * the board.dts only if below settings are supported so that this sample
+ * works.
+ */
+#define ADC_1ST_CHANNEL_ID	0
+#define ADC_RESOLUTION		12
+#define ADC_GAIN		ADC_GAIN_1
+#define ADC_REFERENCE		ADC_REF_INTERNAL
+#define ADC_ACQUISITION_TIME	ADC_ACQ_TIME_DEFAULT
+
+static int16_t m_sample_buffer[ADC_NUM_CHANNELS];
+
+struct adc_channel_cfg channel_cfg = {
+	.gain = ADC_GAIN,
+	.reference = ADC_REFERENCE,
+	.acquisition_time = ADC_ACQUISITION_TIME,
+	/* channel ID will be overwritten below */
+	.channel_id = 0,
+	.differential = 0
+};
+
+struct adc_sequence sequence = {
+	/* individual channels will be added below */
+	.channels    = 0,
+	.buffer      = m_sample_buffer,
+	/* buffer size in bytes, not number of samples */
+	.buffer_size = sizeof(m_sample_buffer),
+	.resolution  = ADC_RESOLUTION,
+};
+
+void main(void)
+{
+	int err;
+	struct device *dev_adc = device_get_binding(ADC_DEVICE_NAME);
+
+	if (!dev_adc) {
+		printk("ADC device not found\n");
+		return;
+	}
+
+	/*
+	 * Configure channels individually prior to sampling
+	 */
+	for (uint8_t ch = ADC_1ST_CHANNEL_ID; ch < ADC_NUM_CHANNELS; ch++) {
+		channel_cfg.channel_id = ch;
+		adc_channel_setup(dev_adc, &channel_cfg);
+
+		sequence.channels |= BIT(ch);
+	}
+
+	while (1) {
+		/*
+		 * Read sequence of channels (if supported by MCU)
+		 */
+		err = adc_read(dev_adc, &sequence);
+		if (err != 0) {
+			printk("ADC reading failed with error %d.\n", err);
+			return;
+		}
+
+		printk("Raw ADC reading(s):");
+		for (uint8_t ch = ADC_1ST_CHANNEL_ID; ch < ADC_NUM_CHANNELS;
+									ch++) {
+			printk(" %d", m_sample_buffer[ch]);
```

Comment:
```suggestion
			printk(" %d", m_sample_buffer[ch - ADC_1ST_CHANNEL_ID]);
```

Reply:
Changed this to just loop from 0 to ADC_NUM_CHANNELS in order to make clear that the values in the buffer start from index 0 independent of the selected channels.

For the other loop I implemented the fix as you suggested. Thanks!